odra-casper/livenet-env/src/livenet_host_env.rs

```diff
@@ -0,0 +1,174 @@
+
+    pub fn new_instance() -> Self {
+        let casper_client: Rc<RefCell<CasperClient>> = Default::default();
+        let callstack: Rc<RefCell<Callstack>> = Default::default();
+        let contract_register = Arc::new(RwLock::new(Default::default()));
+        let livenet_contract_env = LivenetContractEnv::new(
+            casper_client.clone(),
+            callstack.clone(),
+            contract_register.clone()
+        );
+        let contract_env = Rc::new(ContractEnv::new(0, livenet_contract_env));
+        Self {
+            casper_client,
+            contract_register,
+            contract_env,
+            callstack
+        }
+    }
+}
+
+impl HostContext for LivenetEnv {
+    fn set_caller(&self, caller: Address) {
+        self.casper_client.borrow_mut().set_caller(caller);
+    }
+
+    fn set_gas(&self, gas: u64) {
+        self.casper_client.borrow_mut().set_gas(gas);
+    }
+
+    fn caller(&self) -> Address {
+        self.casper_client.borrow().caller()
+    }
+
+    fn get_account(&self, index: usize) -> Address {
+        self.casper_client.borrow().get_account(index)
+    }
+
+    fn balance_of(&self, address: &Address) -> U512 {
+        self.casper_client.borrow().get_balance(address)
+    }
+
+    fn advance_block_time(&self, time_diff: u64) {
+        // Todo: implement logging, especially for such cases:
+        // info!("advance_block_time called - Waiting for {} ms", time_diff);
+        sleep(std::time::Duration::from_millis(time_diff));
+    }
+
+    fn get_event(&self, contract_address: &Address, index: i32) -> Result<Bytes, EventError> {
+        // TODO: handle indices < 0
+        self.casper_client
+            .borrow()
+            .get_event(contract_address, index)
+            .map_err(|_| CouldntExtractEventData)
+    }
+
+    fn get_events_count(&self, contract_address: &Address) -> u32 {
+        self.casper_client.borrow().events_count(contract_address)
+    }
+
+    fn call_contract(
+        &self,
+        address: &Address,
+        call_def: CallDef,
+        use_proxy: bool
+    ) -> Result<Bytes, OdraError> {
+        if !call_def.is_mut() {
+            self.callstack
+                .borrow_mut()
+                .push(CallstackElement::Entrypoint(Entrypoint::new(
+                    *address,
+                    call_def.clone()
+                )));
+            let result = self
+                .contract_register
+                .read()
+                .unwrap()
+                .call(address, call_def);
+            self.callstack.borrow_mut().pop();
+            return result;
+        }
+        match use_proxy {
+            true => self
+                .casper_client
+                .borrow_mut()
+                .deploy_entrypoint_call_with_proxy(*address, call_def),
+            false => {
+                self.casper_client
+                    .borrow_mut()
+                    .deploy_entrypoint_call(*address, call_def);
+                Ok(Default::default())
```

Comment:
I think you should return `()::to_bytes()` or sth like that so it matches when:
```rust
let result: () = env.call_contract(...).unwrap()
````